wgpu-native/src/command/compute.rs

```diff
@@ -144,34 +144,33 @@ pub extern "C" fn wgpu_compute_pass_set_pipeline(
         pass.raw.bind_compute_pipeline(&pipeline.raw);
     }
 
-    if pass.binder.pipeline_layout_id == Some(pipeline.layout_id.clone()) {
-        return;
-    }
-
-    let pipeline_layout_guard = HUB.pipeline_layouts.read();
-    let pipeline_layout = &pipeline_layout_guard[pipeline.layout_id];
-    let bind_group_guard = HUB.bind_groups.read();
-
-    pass.binder.pipeline_layout_id = Some(pipeline.layout_id.clone());
-    pass.binder
-        .reset_expectations(pipeline_layout.bind_group_layout_ids.len());
+    // Rebind resources
+    if pass.binder.pipeline_layout_id != Some(pipeline.layout_id.clone()) {
```

Comment:
How did the early out functionally change anything here? Or is this to make rebinding more explicit?

Reply:
the compute side was fine, but I refactored it to be in line with the render side changes